src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -0,0 +1,259 @@
+    if (!this.expanded) {
+      this.open();
+    } else if (this.expanded) {
+      this.close();
+    }
+  }
+
+  private handleClick(e: Event) {
+    if (e.target == this.element || e.target == $$(this.element).find('.coveo-simplefilter-select') ||
+      e.target == $$(this.element).find('.coveo-circle') || e.target == $$(this.element).find('.coveo-icon')
+      || e.target == $$(this.element).find('.coveo-simplefilter-selecttext')) {
+
+      e.stopImmediatePropagation();
+      this.toggle();
+    }
+  }
+
+
+  private open() {
+    $$(this.element).addClass('coveo-checkbox-container-expanded');
+    this.checkboxContainer.addClass('coveo-checkbox-container-expanded');
+    if ($$(document.body).find('.coveo-dropdown-background') == null) {
+      let backdrop = $$('div', { className: 'coveo-dropdown-background' });
+      backdrop.on('click', () => this.close());
+      this.element.parentElement.appendChild(backdrop.el);
+    } else {
+      const dropdown = $$(document.body).find('.coveo-dropdown-background');
+      $$(dropdown).on('click', () => this.close());
+    }
+  }
+
+  private close() {
+    $$(this.element).removeClass('coveo-checkbox-container-expanded');
+    this.checkboxContainer.removeClass('coveo-checkbox-container-expanded');
+    if ($$(this.element.parentElement).find('.coveo-dropdown-background') != null && $$(document.body).find('.coveo-checkbox-container-expanded') == null) {
+      $$(this.element.parentElement).find('.coveo-dropdown-background').remove();
+    }
+  }
+
+  private handleCheckboxToggle() {
+    this.circleElement.text(this.getSelectedValues().length.toString());
+    const title = $$(this.element).find('.coveo-simplefilter-selecttext');
+    if (this.getSelectedValues().length == 1) {
+      $$(title).text(l(this.getSelectedValues()[0]));
+    } else {
+      $$(title).text(this.options.title);
+    }
+    this.queryController.executeQuery();
+  }
+
+  private createCheckboxes(value: string) {
+    const checkbox = new Checkbox(() => {
+      this.handleCheckboxToggle();
+    }, l(this.getValueCaption(value)));
+    return { checkbox, value };
+  }
+
+  public getValueCaption(value: string): string {
+    let ret = value;
+
+    if (Utils.exists(this.options.valueCaption)) {
+      if (typeof this.options.valueCaption == 'object') {
+        ret = this.options.valueCaption[ret] || ret;
+      }
+    }
+    return ret;
+  }
+
+  public getSelected(): { checkbox: Checkbox, value: string }[] {
+    return _.filter(this.checkboxes, (object: { checkbox: Checkbox, value: string }) => object.checkbox.isSelected());
+  }
+
+  public getSelectedValues(): string[] {
+    return _.map(this.getSelected(), (object: { checkbox: Checkbox, value: string }) => object.value);
+  }
+
+  private buildCircleElement(): Dom {
+    this.circleElement = $$('span', { className: 'coveo-circle' }, this.getSelectedValues().length.toString());
+    return this.circleElement;
+  }
+
+  private createWrapper() {
```

Comment:
I'd call this `findOrCreate` wrapper.

Also, you should always be searching from `this.root`, not `document.body`.

What happens if there is 2 search interface in a single page ? Things would get weird because both search interface would pollute each other.

By scoping on "root", you eliminate those kind of problem. `root` is the `div class="CoveoSearchInterface"` inside which you component exist.